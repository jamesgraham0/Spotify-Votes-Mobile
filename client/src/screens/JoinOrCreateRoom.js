import { StyleSheet, Text, View, TouchableOpacity } from 'react-native';
import React from 'react';
import * as Haptics from 'expo-haptics';

const JoinOrCreateRoom = ({ navigation, route }) => {
    const { user } = route.params;
    const name = user.display_name;
    const isPremiumAccount = user.product !== "free";

    const handleCreateRoom = () => {
        navigation.navigate('CreateRoom', { user: user });
    }

    const handleJoinRoom = () => {
        navigation.navigate('JoinRoom', { user: user });
    }

    return (
        <View style={styles.container}>
            <Text style={styles.title}> Spotify Votes </Text>
            <Text style={styles.HiMessage}> Hi, {name}! </Text>
            <TouchableOpacity 
                disabled={!isPremiumAccount}
                onPress={() => {
                    Haptics.impactAsync(Haptics.ImpactFeedbackStyle.Heavy); 
                    handleCreateRoom();
                }} 
                style={!isPremiumAccount ? styles.createRoomButtonDisabled : styles.createRoomButton}>
                    {isPremiumAccount ? 
                        <Text style={styles.createRoomText}>Create Room</Text> 
                        :
                        <Text style={styles.createRoomTextDisabled}>Need premium account to create room</Text>
                    }
            </TouchableOpacity>
            <TouchableOpacity 
            onPress={() => {
                Haptics.impactAsync(Haptics.ImpactFeedbackStyle.Heavy); 
                handleJoinRoom();
            }} 
            style={styles.joinRoomButton}>
                <Text style={styles.joinRoomText}>Join Room</Text>
            </TouchableOpacity>
    </View>
    )
}

const styles = StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: '#191414',
      alignItems: 'center',
    },
    title: {
      top: 100,
      fontSize: 40,
      color: '#1DB954',
    },
    HiMessage: {
        top: 200,
        color: '#fff',
        fontSize: 30,
    },
    createRoomButtonDisabled: {
        top: 400,
        borderWidth: 2,
        borderColor: '#444',
        width: 300,
        height: 80,
        borderRadius: 50,
        shadowOpacity: 1,
        shadowColor: '#444',
        shadowOffset: { width: 0, height: 0 },
        shadowRadius: 5,
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
    },
    createRoomButton: {
        top: 400,
        borderWidth: 2,
        borderColor: '#B026FF',
        width: 300,
        height: 80,
        borderRadius: 50,
        shadowOpacity: 1,
        shadowColor: '#B026FF',
        shadowOffset: { width: 0, height: 0 },
        shadowRadius: 5,
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
    },
    joinRoomButton: {
        marginTop: 25,
        top: 400,
        borderWidth: 2,
        borderColor: '#1DB954',
        width: 300,
        height: 80,
        borderRadius: 50,
        shadowOpacity: 1,
        shadowColor: '#1DB954',
        shadowOffset: { width: 0, height: 0 },
        shadowRadius: 5,
        color: '#1DB954',
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
    },
    createRoomTextDisabled: {
        position: 'absolute',
        color: '#444',
        textAlign: 'center',
        fontSize: 20,
    },
    createRoomText: {
        position: 'absolute',
        color: '#fff',
        fontSize: 24,
    },
    joinRoomText: {
        position: 'absolute',
        color: '#fff',
        fontSize: 24,
    }
  })
  

export default JoinOrCreateRoom;