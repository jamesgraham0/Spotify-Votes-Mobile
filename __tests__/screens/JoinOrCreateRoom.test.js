import React from 'react';
import renderer from 'react-test-renderer';
import JoinOrCreateRoom from '../../client/src/screens/JoinOrCreateRoom';

describe('JoinOrCreateRoom', () => {
    it('should render the JoinOrCreateRoom component', () => {
        const navigation = {};
        const route = { params: { user: {} } };
        const tree = renderer.create(<JoinOrCreateRoom navigation={navigation} route={route} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});