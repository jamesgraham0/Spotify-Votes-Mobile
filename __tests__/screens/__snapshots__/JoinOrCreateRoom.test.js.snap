// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JoinOrCreateRoom should render the JoinOrCreateRoom component 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#191414",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "height": "100%",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      style={
        [
          {
            "borderRadius": 125,
            "elevation": 5,
            "height": 250,
            "opacity": 0.4,
            "position": "absolute",
            "right": -60,
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0.8,
            "shadowRadius": 6,
            "top": -60,
            "width": 250,
          },
          {
            "backgroundColor": "#1DB954",
            "shadowColor": "#1DB954",
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "borderRadius": 25,
            "elevation": 5,
            "height": 50,
            "opacity": 0.4,
            "position": "absolute",
            "right": 100,
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0.8,
            "shadowRadius": 6,
            "top": 150,
            "width": 50,
          },
          {
            "backgroundColor": "#1DB954",
            "shadowColor": "#1DB954",
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "borderRadius": 45,
            "elevation": 5,
            "height": 90,
            "opacity": 0.4,
            "position": "absolute",
            "right": 130,
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0.8,
            "shadowRadius": 6,
            "top": 110,
            "width": 90,
          },
          {
            "backgroundColor": "#1DB954",
            "shadowColor": "#1DB954",
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "borderRadius": 75,
            "bottom": -20,
            "elevation": 5,
            "height": 150,
            "left": -20,
            "opacity": 0.4,
            "position": "absolute",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0.8,
            "shadowRadius": 6,
            "width": 150,
          },
          {
            "backgroundColor": "#1DB954",
            "shadowColor": "#1DB954",
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "borderRadius": 50,
            "bottom": 100,
            "elevation": 5,
            "height": 100,
            "left": -40,
            "opacity": 0.4,
            "position": "absolute",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0.8,
            "shadowRadius": 6,
            "width": 100,
          },
          {
            "backgroundColor": "#1DB954",
            "shadowColor": "#1DB954",
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "borderRadius": 75,
            "elevation": 5,
            "height": 150,
            "left": -10,
            "opacity": 0.4,
            "position": "absolute",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0.8,
            "shadowRadius": 6,
            "top": -20,
            "width": 150,
          },
          {
            "backgroundColor": "#B026FF",
            "shadowColor": "#B026FF",
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "borderRadius": 40,
            "elevation": 5,
            "height": 80,
            "left": -10,
            "opacity": 0.4,
            "position": "absolute",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0.8,
            "shadowRadius": 6,
            "top": 100,
            "width": 80,
          },
          {
            "backgroundColor": "#B026FF",
            "shadowColor": "#B026FF",
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "borderRadius": 15,
            "elevation": 5,
            "height": 30,
            "left": 120,
            "opacity": 0.4,
            "position": "absolute",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0.8,
            "shadowRadius": 6,
            "top": 30,
            "width": 30,
          },
          {
            "backgroundColor": "#B026FF",
            "shadowColor": "#B026FF",
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "borderRadius": 75,
            "bottom": -20,
            "elevation": 5,
            "height": 150,
            "opacity": 0.4,
            "position": "absolute",
            "right": -20,
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0.8,
            "shadowRadius": 6,
            "width": 150,
          },
          {
            "backgroundColor": "#B026FF",
            "shadowColor": "#B026FF",
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "borderRadius": 50,
            "bottom": 100,
            "elevation": 5,
            "height": 100,
            "opacity": 0.4,
            "position": "absolute",
            "right": -40,
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0.8,
            "shadowRadius": 6,
            "width": 100,
          },
          {
            "backgroundColor": "#B026FF",
            "shadowColor": "#B026FF",
          },
        ]
      }
    />
  </View>
  <Text
    style={
      {
        "color": "#1DB954",
        "fontSize": 40,
        "top": 230,
      }
    }
  >
     Spotify Votes 
  </Text>
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 30,
        "top": 280,
      }
    }
  >
     Hi, 
    ! 
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#191414",
        "borderColor": "#B026FF",
        "borderRadius": 50,
        "borderWidth": 2,
        "display": "flex",
        "height": 80,
        "justifyContent": "center",
        "opacity": 1,
        "shadowColor": "#B026FF",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 5,
        "top": 350,
        "width": 300,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#fff",
            "fontSize": 24,
            "position": "absolute",
          },
          {
            "color": "#fff",
            "fontSize": 24,
            "position": "absolute",
          },
        ]
      }
    >
      Create Room
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#191414",
        "borderColor": "#1DB954",
        "borderRadius": 50,
        "borderWidth": 2,
        "color": "#1DB954",
        "display": "flex",
        "height": 80,
        "justifyContent": "center",
        "marginTop": 25,
        "opacity": 1,
        "shadowColor": "#1DB954",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 5,
        "top": 350,
        "width": 300,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#fff",
            "fontSize": 24,
            "position": "absolute",
          },
          {
            "color": "#fff",
            "fontSize": 24,
            "position": "absolute",
          },
        ]
      }
    >
      Join Room
    </Text>
  </View>
</View>
`;
